app/controllers/api/v2/instance_hosts_controller.rb

```diff
@@ -0,0 +1,32 @@
+module Api
+  module V2
+    class InstanceHostsController < V2::BaseController
+      def resource_class
+        @resource_class ||= Host::Managed
+      end
+
+      def resource_scope(*args)
+        super.authorized(:view_hosts).joins(:infrastructure_facet).merge(::HostFacets::InfrastructureFacet.where(foreman: true))
+      end
+
+      api :PUT, '/instance/hosts/:host_id', N_("Assign a host to the Foreman instance")
```

Comment:
Will there be other things potentially rooted at `/instance` in the future? I was thinking about why it is `/instance/hosts` vs. `/instance_hosts`.

Reply:
Since we may want to use this for eventually managing foreman from foreman, all the operations could live under the `/instance` root.

Reply:
Makes sense. I think stand-alone `instance` term just worries because we are also a project that manages virtual machines and compute resources where `instance` is often a term used to reference a created VM. Whereas `foreman_instance` is a bit more explicit. If others like @evgeni @ekohl and yourself are not worried then ignore my commentary.

Reply:
I wonder if `self` would be clear enough. So `/self/hosts/:host_id`. Similar we could have `/self/status`, `/self/features` or `/self/plugins` or whatever.